presto-main/src/main/java/com/facebook/presto/execution/QueryExecution.java

```diff
@@ -34,6 +34,8 @@ Duration waitForStateChange(QueryState currentState, Duration maxWait)
 
     long getTotalMemoryReservation();
 
+    long getMaxQueryMemory(long defaultValue);
```

Comment:
Instead of adding this, I would add a `getSession()` method, then you can use that directly in `ClusterMemoryManager`